### PR TITLE
2: remove discussions backend code, reddit, channels, moira, notifications

### PR DESCRIPTION
**Pull request by collinpreston**
#### Pre-Flight checklist
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/2

#### What's this PR do?

1. Removes all Reddit related functionality (praw, posts, channels, comments, notifications, etc).
2. Removes Moira related functionality.
3. Removes some remaining auth functionality that will be replaced by Keycloak.
4. Removes unused environment variables.
5. Removes Akismet which was used for mitigating spam comments in Reddit.

#### How should this be manually tested?

1. All tests should pass.
2. All MIT Open related functionality perform the same as the current.
3. Review modified to code and management commands.

#### Questions for reviewer

1. Should ckeditor remain or be removed?
2. Should uploading to S3 functionality remain or be removed?

### CODE REVIEW

File: embedly/views.py [hunk 1]
```diff
@@ -6,8 +6,7 @@
 
 from django.conf import settings
 
-from channels.models import LinkMeta
-from embedly.api import get_embedly_summary, THUMBNAIL_URL
```
Comment (mbertrand) [inline embedly/views.py:10]: You can probably delete the constant `THUMBNAIL_URL` from `embedly.api`

File: search/management/commands/update_index.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def add_arguments(self, parser):
         super().add_arguments(parser)
 
     def handle(self, *args, **options):
-        """Index the comments and posts for the channels the user is subscribed to"""
+        """Index Courses"""
```
Comment (mbertrand) [inline search/management/commands/update_index.py:50]: """Index all learning resources"""?

**Review by mbertrand — CHANGES_REQUESTED** [review]
> Should ckeditor remain or be removed?

I'd defer to @ChristopherChudzicki on that, but I think there's a chance we might want it for field channels and/or their widgets?

> Should uploading to S3 functionality remain or be removed?

Are you referring to `MITOPEN_USE_S3` and `DEFAULT_FILE_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"`?  I think we should keep that, for when FieldChannel images are uploaded, and maybe user profile images too (unless keycloak will be replacing that part).

**Comment by ChristopherChudzicki** [comment]
> > Should ckeditor remain or be removed?

>I'd defer to @ChristopherChudzicki on that, but I think there's a chance we might want it for field channels and/or their widgets?

I would keep it. The API endpoint gives us a key for CKEditor's image upload service, which we may well continue to use.

**Review by mbertrand — APPROVED** [review]
LGTM! 👍

**Pull request merged**